lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -0,0 +1,890 @@
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function minute($expression)
+    {
+        return $this->operator('$minute', $expression);
+    }
+
+    /**
+     * Divides one number by another and returns the remainder. The first argument is divided by the second argument.
+     *
+     * The arguments can be any valid expression as long as they resolve to numbers.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/mod/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function mod($expression1, $expression2)
+    {
+        return $this->operator('$mod', array($expression1, $expression2));
+    }
+
+    /**
+     * Returns the month of a date as a number between 1 and 12.
+     *
+     * The argument can be any expression as long as it resolves to a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/month/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function month($expression)
+    {
+        return $this->operator('$month', $expression);
+    }
+
+    /**
+     * Multiplies numbers together and returns the result.
+     *
+     * The arguments can be any valid expression as long as they resolve to numbers.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/multiply/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @param mixed|self $expression3,... Additional expressions
+     * @return self
+     */
+    public function multiply($expression1, $expression2 /* , $expression3, ... */)
+    {
+        return $this->operator('$multiply', func_get_args());
+    }
+
+    /**
+     * Compares two values and returns:
+     * true when the values are not equivalent.
+     * false when the values are equivalent.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/ne/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function ne($expression1, $expression2)
+    {
+        return $this->operator('$ne', array($expression1, $expression2));
+    }
+
+    /**
+     * Evaluates a boolean and returns the opposite boolean value.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/not/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function not($expression)
+    {
+        return $this->operator('$not', $expression);
+    }
+
+    /**
+     * Defines an operator and value on the expression.
+     *
+     * If there is a current field, the operator will be set on it; otherwise,
+     * the operator is set at the top level of the query.
+     *
+     * @param string $operator
+     * @param array|self[]|self $expression
+     * @return self
+     */
+    protected function operator($operator, $expression)
```

Comment:
I think making this method public would expose too much functionality to the end user. The aggregation builder is meant to guide the user when writing aggregation queries, similar to the query builder. Allowing a "just put whatever you want in here" is a bit dangerous if you ask me.
The ```expression()``` method was added to allow creating creating nested documents in a ```$group``` or ```$project``` stage. That's also the reason why it requires ```field()``` to be called first.

Also, while the query builder uses the field for querying purposes, it's important to remember that most operators can be called without a field. The ```field()``` method defines the field where the result of an expression will be stored. Take this example:
```
$builder
    ->project()
    ->field('totalAmount')
    ->multiply('$taxRate', $this->expr()->subtract('$price', '$rebate'));
```
In this case, one operator is applied with a previous call to ```field()``` while the other operator is applied without calling ```field()``` because we don't need to store its value, we're just using it in the other expression.